src/intrinsics/ecma262/Symbol.js

```diff
@@ -28,11 +28,12 @@ export default function(realm: Realm): NativeFunctionValue {
     let descString;
     if (!description || description instanceof UndefinedValue) {
       descString = undefined;
+    } else if (description instanceof AbstractValue) {
+      return new SymbolValue(realm, description);
```

Comment:
+1 for @yinghuitan comment.

Also, this will skip the ToString(Partial) call, which is okay only if the AbstractValue is known to be a string. So, also guard this case by  ` && !description.mightNotBeString()`.